components/scream/src/physics/shoc/shoc_pblintd_surf_temp_impl.hpp

```diff
@@ -13,10 +13,35 @@ namespace shoc {
 
 template<typename S, typename D>
 KOKKOS_FUNCTION
-void Functions<S,D>::pblintd_surf_temp(const Int& shcol, const Int& nlev, const Int& nlevi, const uview_1d<const Spack>& z, const uview_1d<const Spack>& ustar, const uview_1d<const Spack>& obklen, const uview_1d<const Spack>& kbfs, const uview_1d<const Spack>& thv, const uview_1d<Spack>& tlv, const uview_1d<Spack>& pblh, const uview_1d<bool>& check, const uview_1d<Spack>& rino)
+void Functions<S,D>::pblintd_surf_temp(const Int& nlev, const Int& nlevi, const Int& npbl,
+      const uview_1d<const Spack>& z, const Scalar& ustar, 
+      const Scalar& obklen, const Scalar& kbfs, 
+      const uview_1d<const Spack>& thv, Scalar& tlv, 
+      Scalar& pblh, bool& check, const uview_1d<Spack>& rino)
 {
-  // TODO
-  // Note, argument types may need tweaking. Generator is not always able to tell what needs to be packed
+  // const parameter for Diagnosis of PBL depth
+  const auto onet  = 1./3.;
+  const auto fak   =  8.5;
+  const auto betam = 15.0;
+  const auto sffrac=  0.1;
```

Comment:
These need to be declared as Scalars. The tests are failing in SP and not DP, so my guess is that auto chooses double instead of float.